Runtime/Scripts/Export/GltfWriter.cs

```diff
@@ -1675,43 +1800,51 @@ NativeArray<byte> outputStream
             return node;
         }
         
-        int AddMesh(UnityEngine.Mesh uMesh) {
-            int meshId;
-            
+        void AddMeshAndSkin(UnityEngine.Mesh uMesh, Transform[] bones, out int meshId, out int skinId)
+        {
+            meshId = -1;
+            skinId = -1;
 #if !UNITY_EDITOR
             if (!uMesh.isReadable) {
                 m_Logger?.Error(LogCode.MeshNotReadable, uMesh.name);
-                return -1;
+                return;
             }
 #endif
        
-            if (m_UnityMeshes!=null) {
-                meshId = m_UnityMeshes.IndexOf(uMesh);
-                if (meshId >= 0) {
-                    return meshId;
-                }
+            meshId = m_UnityMeshes?.IndexOf(uMesh) ?? -1;
+            if (meshId == -1) {
+                var mesh = new Mesh {
+                    name = uMesh.name
+                };
+                m_Meshes = m_Meshes ?? new List<Mesh>();
+                m_UnityMeshes = m_UnityMeshes ?? new List<UnityEngine.Mesh>();
+                m_Meshes.Add(mesh);
+                m_UnityMeshes.Add(uMesh);
+                meshId = m_Meshes.Count - 1;
             }
 
-            var mesh = new Mesh {
-                name = uMesh.name
-            };
-            m_Meshes = m_Meshes ?? new List<Mesh>();
-            m_UnityMeshes = m_UnityMeshes ?? new List<UnityEngine.Mesh>();
-            m_Meshes.Add(mesh);
-            m_UnityMeshes.Add(uMesh);
-            meshId = m_Meshes.Count - 1;
-            return meshId;
+            m_Skins = m_Skins ?? new List<Skin>();
+            m_uBones = m_uBones ?? new List<Transform[]>();
+            m_SkinMap = m_SkinMap ?? new Dictionary<UnityEngine.Mesh, Skin>();
+            if (!m_SkinMap.ContainsKey(uMesh))
+            {
+                m_Skins.Add(new Skin());
+                m_SkinMap[uMesh] = m_Skins.Last();
+                m_uBones.Add(bones);
+            }
+
+            skinId = m_Skins.IndexOf(m_SkinMap[uMesh]);

```

Comment:
I'd make this similar than the meshId query above. Start with `m_Skins.IndexOf` and only if there's no no hit (and `bones` is valid ofc) create stuff.
Why:
- no need to initialize m_Skins etc. upfront
- Querying `m_Skins` just once instead of twice